src/sdp_proxy/advertising_proxy.cpp

```diff
@@ -147,6 +147,11 @@ void AdvertisingProxy::AdvertisingHandler(otSrpServerServiceUpdateId aId,
 
     error = PublishHostAndItsServices(aHost, update);
 
+    if (error == OTBR_ERROR_ABORTED)
```

Comment:
@wgtdkp Specifically, could you share your thoughts on this change?

Reply:
If we want to ignore the advertising proxy error, please do it on the SRP server side. The Advertising Proxy shouldn't know the the caller context but should always reports any error encountered.

Reply:
Personally, I think the SRP server should wait for the first SRP update message being processed completely before sending the second SRP update message for the same host, This will eliminate the ABORTED error entirely. @abtink thoughts?

Reply:
In general, I agree we should not convert the error here.

But I don't understand why do we need it?

Reply:
Is it that when we do SRPL initial sync, we may get a bunch of messages for same host at the same time and then each will call proxy?

Reply:
Yes there may be an issue during initial sync.

Say that there are two BRs. 

1. BR1 gets an SRP update for adding `service1` and `service2` for the `host1`.
2. BR1 gets an SRP update for removing `service1`. BR2 should still have `service2` registered.
3. Boot BR2. BR2 starts initial sync with BR1. BR2's Advertising Proxy will publish the `service1`, `service2` and unpublish `service1` immediately. In this way the Advertising Proxy will get an `Aborted` error and mark the first SRP update as `Aborted`. In this way `service2` won't be registered as SRP service on BR2.

Reply:
I see. Thanks. I agree this may be better handled on SRP server code. Ideally we should wait till end of processing an SRPL sync before publishing anything with adv proxy. But I am not sure how easy/hard it may be.

So for now, let's keep this in this PR.

Reply:
After a second thought, I think it may be better to do this workaround (`Aborted`->`None`) at [SRP server side](https://github.com/openthread/openthread/blob/d28793c104ab710f692a160377a5753373ae4b69/src/core/net/srp_server.cpp#L405). The SRP server has the information whether the update message is from a client or an SRPL partner so we can limit the impact of this workaround.

Reply:
Yes, I don't it's a good idea to ignore this advertising proxy error for event a direct SRP client registration

Reply:
Added this in SRPL PR on OT core ->https://github.com/openthread/openthread/pull/8403/commits/5364202199856832b6a42f492ef44d9e66e26fe2